content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -0,0 +1,284 @@
+---
+title: "Going From Ember to Vue"
+date: "2019-08-03T17:02:38.041Z"
+template: "post"
+draft: false
+slug: "/posts/going-from-ember-to-vue/"
+category: "Engineering"
+banner: "https://miro.medium.com/max/3840/1*nfvapd86apvGH-hNBYkYuw.png"
+filename: "Going-from-ember-to-vue.md"
+tags:
+  - "Engineering"
+  - "Ember"
+  - "Vue"
+  - "JavaScript"
+  - "Web Development"
+description: "Thinking of looking into another framework other than Ember? React might seem like the obvious choice due to popularity, but consider taking a look at Vue first!"
+---
+
+![Vue.js logo](https://miro.medium.com/max/3840/1*nfvapd86apvGH-hNBYkYuw.png)
+
+If you know me, you know that I love Ember and have worked with it (and the community) for several years.
+But recently, I took another job where I won't be writing it anymore [(I wrote a whole post about this if you're interested)](/posts/a-few-thoughts-on-ember/).
+My first instinct when looking for a new job was to learn React, due to popularity.
+But I found it was hard to jump straight from Ember into React; not impossible, but I felt I had to unlearn some habits that I had formed.
+For the past few months though, I have been working on a Vue project and I have found the leap from Ember to Vue was much more straight forward.
+
+So, I was going to take some time and write about the similarities and differences I have experienced.
+Even if you still love Ember and plan on working with it for a while, I would encourage you to look at Vue just to see what else is out there.
+
+# Some big differences
+
+Much like React, Vue is a component library -- which means you'll be using components for <strong>everything.</strong>
+That means no controllers or route objects.
+When you create a route, you are pointing to a component.
+There is literally nothing special about this component -- it can be literally any component that you want (in fact, you can use it both as a route _and_ component on the page... although, you really shouldn't).
+
+In my opinion, this is a good habit to get into as it will help you transition to something like React down the line if you have to.
```

Comment:
what about testing?  Would be curious on the testing apparatus and how you like it :)

Reply:
I went back and forth on whether to include testing stuff, but that's a whole blog post in itself.

Much like routing and state management, it doesn't come out of the box but there is an officially supported solution:
https://vue-test-utils.vuejs.org/guides/getting-started.html
Using Jest as the test runner is what I've seen most, but you can use mocha as well.

From what I've done, it's very specific component-level testing.  I haven't really done Acceptance level testing in a project yet

Reply:
Does the Vue community encourage testing against a real browser? I think the jest default is jsdom, which is all fake, and users don't use fake browsers.

Reply:
I hope it's gonna be even better since mirage is being extracted from `ember-cli-mirage`

Reply:
> Does the Vue community encourage testing against a real browser? I think the jest default is jsdom, which is all fake, and users don't use fake browsers.

From what I have seen so far, most of the documentation and examples are with `jest` and not real browser testing. That being said, I've only worked on 3 Vue apps with very similar approaches, so I'm sure examples exists somewhere.